docs/middleware.rst

```diff
@@ -1,34 +1,34 @@
+.. _internals__middlewares:
```

Comment:
```suggestion
.. _internals__middleware:
```

Keeping consistency, we should change this to middleware and update wherever in the docs this link is referenced as well.